src/fdb5/tools/fdb-list.cc

```diff
@@ -29,54 +29,47 @@
 using namespace eckit;
 using namespace eckit::option;
 
-namespace fdb5 {
-namespace tools {
+namespace fdb5::tools {
 
 //----------------------------------------------------------------------------------------------------------------------
 
-class FDBList : public FDBVisitTool {
-
-  public: // methods
-
-    FDBList(int argc, char **argv) :
-        FDBVisitTool(argc, argv, "class,expver"),
-        location_(false),
-        timestamp_(false),
-        length_(false),
-        full_(false),
-        porcelain_(false),
-        json_(false) {
-
+class FDBList: public FDBVisitTool {
+public:  // methods
+    FDBList(int argc, char** argv): FDBVisitTool(argc, argv, "class,expver") {
         options_.push_back(new SimpleOption<bool>("location", "Also print the location of each field"));
         options_.push_back(new SimpleOption<bool>("timestamp", "Also print the timestamp when the field was indexed"));
         options_.push_back(new SimpleOption<bool>("length", "Also print the field size"));
         options_.push_back(new SimpleOption<bool>("full", "Include all entries (including masked duplicates)"));
-        options_.push_back(new SimpleOption<bool>("porcelain", "Streamlined and stable output for input into other tools"));
+        options_.push_back(new SimpleOption<bool>("porcelain",
+                                                  "Streamlined and stable output. "
+                                                  "Useful as input for other tools or scripts"));
         options_.push_back(new SimpleOption<bool>("json", "Output available fields in JSON form"));
         options_.push_back(new SimpleOption<bool>("compact", "Aggregate available fields in MARS requests"));
+        options_.push_back(new SimpleOption<long>("depth", "Output entries up to 'depth' levels deep"));
     }
 
-  private: // methods
-
-    virtual void execute(const CmdArgs& args);
-    virtual void init(const CmdArgs &args);
-
-    bool location_;
-    bool timestamp_;
-    bool length_;
-    bool full_;
-    bool porcelain_;
-    bool json_;
-    bool compact_;
+private:  // methods
+    void execute(const CmdArgs& args) override;
+    void init(const CmdArgs& args) override;
+
+    bool location_ {false};
+    bool timestamp_ {false};
+    bool length_ {false};
+    bool full_ {false};
+    bool porcelain_ {false};
+    bool json_ {false};
+    bool compact_ {false};
+    int  depth_ {3};
 };
 
+//----------------------------------------------------------------------------------------------------------------------
 
 std::string keySignature(const fdb5::Key& key) {
     std::string signature;
     std::string separator;
-    for (auto k : key.keys()) {
-        signature += separator+k;
-        separator=":";
+    for (auto&& k : key.keys()) {
```

Comment:
Should be "const auto&". Doesn't need && - not intending either to do move(), perfect forwarding or lifetime extension.

Reply:
unrelated perf fix.. here, "auto&&" becomes "const auto&"